recipes/c-client/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+import functools
+import os
+import stat
+
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class CclientConan(ConanFile):
+    name = "c-client"
+    description = "University of Washington IMAP toolkit"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/uw-imap/imap"
+    topics = "imap"
+    license = "Apache-2.0"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+    exports_sources = "patches/*"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    @property
+    def _is_msvc(self):
+        return self._settings_build.compiler in ("Visual Studio", "msvc")
+
+    def validate(self):
+        if self._settings_build.os == "Windows" and not self._is_msvc:
+            raise ConanInvalidConfiguration(
+                "c-client is setup to build only with MSVC on Windows"
+            )
+        # FIXME: need krb5 recipe
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration(
+                "c-client depends on krb5 on MacOS and it's not packaged by "
+                "Conan yet"
+            )
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if not self._is_msvc:
+            self.requires("openssl/1.1.1m")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+
+    def _patch_msvc(self):
+        opt_flags = "/O2 /Ob2 /DNDEBUG"
+        if self.settings.build_type == "Debug":
+            opt_flags = "/Zi /Ob0 /Od /RTC1"
+        runtime = f"/{self.settings.compiler.runtime}"
+        # NOTE: boatloads of warnings for truncation, sign mismatch,
+        #       implicit conversions, just the usual C things
+        warnings = \
+            "/W3 /wd4267 /wd4244 /wd4273 /wd4311 /wd4312 /wd4133 /wd4028"
+        cflags = f"{runtime} {warnings} /GS {opt_flags}"
+        search = "EXTRACFLAGS ="
+        replace = f"EXTRACFLAGS = {cflags}"
+        tools.replace_in_file(r"src\osdep\nt\makefile.w2k", search, replace)
+
+    def _build_msvc(self):
+        make = "nmake /nologo /f makefile.w2k"
+        with tools.vcvars(self):
+            self.run(f"{make} c-client")
+            self.run(make, cwd="c-client")
```

Comment:
```suggestion
            self.run(f"{make} c-client", run_environment=True)
            self.run(make, cwd="c-client", run_environment=True)
```


https://docs.conan.io/en/latest/reference/build_helpers/run_environment.html#runenvironment